ngraph/frontend/onnx_import/src/op/reduce.cpp

```diff
@@ -31,7 +32,64 @@ namespace ngraph
         {
             namespace
             {
-                std::shared_ptr<default_opset::Constant> get_reduction_axes(const Node& node)
+                std::shared_ptr<ngraph::Node> get_dynamic_all_axes_range(const Node& node)
+                {
+                    const auto input = node.get_ng_inputs().at(0);
+                    const auto shape_of_input = std::make_shared<default_opset::ShapeOf>(input);
+                    const auto scalar =
+                        default_opset::Constant::create(element::i64, Shape{1}, {0});
+                    const auto rank_of_input =
+                        std::make_shared<default_opset::ShapeOf>(shape_of_input);
+                    const auto rank_of_input_scalar =
+                        std::make_shared<default_opset::Squeeze>(rank_of_input, scalar);
+                    const auto start = default_opset::Constant::create(element::i64, Shape{}, {0});
+                    const auto step = default_opset::Constant::create(element::i64, Shape{}, {1});
+                    return std::make_shared<default_opset::Range>(
+                        start, rank_of_input_scalar, step, element::i64);
+                }
+
+                std::shared_ptr<ngraph::Node> get_reduction_axes_from_input(const Node& node)
+                {
+                    const std::int64_t noop_with_empty_axes =
+                        node.get_attribute_value<std::int64_t>("noop_with_empty_axes", 0);
+                    const auto input = node.get_ng_inputs().at(0);
+                    const auto input_rank = node.get_ng_inputs().at(0).get_partial_shape().rank();
+                    if (node.get_ng_inputs().size() > 1)
+                    {
+                        const auto reduction_axes = node.get_ng_inputs().at(1);
+                        const auto reduction_axes_rank = reduction_axes.get_partial_shape().rank();
+                        NGRAPH_CHECK(reduction_axes.get_partial_shape().is_static(),
+                                     "The axes tensor's shape needs to be known(static). Node: ",
+                                     node.get_description());
+
+                        if (reduction_axes_rank.get_length() != 0 &&
+                            reduction_axes.get_shape() != Shape{0})
+                        {
+                            return reduction_axes.get_node_shared_ptr();
+                        }
+                    }
+
+                    if (noop_with_empty_axes)
+                    {
+                        return nullptr;
+                    }
+                    else
+                    {
+                        if (input_rank.is_static())
```

Comment:
I'm not entirely sure about it but I think we should now always use the `ShapeOf` approach. If we create a constant for a given model, we won't be able to reshape it afterwards.

Reply:
We have discussed it offline and decided to address it in a separated ticket, because this change requires to adapt other operators (i.e. TopK).